File: glyph/src/main/java/com/hsj86715/ingress/glyph/pages/GlyphPairsAdapter.java
package com.hsj86715.ingress.glyph.pages;

import com.hsj86715.ingress.glyph.view.BaseRecyclerAdapter;
import com.hsj86715.ingress.glyphres.data.GlyphInfo;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by hushujun on 2017/5/16.
 * @author hushujun
 */

public class GlyphPairsAdapter extends BaseRecyclerAdapter {
    private List<GlyphInfo[]> mGlyphPairs;

    public GlyphPairsAdapter(List<GlyphInfo[]> glyphPairs) {
        if (glyphPairs == null) {
            this.mGlyphPairs = new ArrayList<>();
        } else {
            this.mGlyphPairs = glyphPairs;
        }
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        GlyphInfo[] infos = mGlyphPairs.get(position);
        holder.glyphPreView.setSequences(infos);
        holder.glyphPreView.setHackSequenceListener(mClickListener);
    }

    @Override
    public int getItemCount() {
        return mGlyphPairs.size();
    }

}

File: app/src/main/java/com/hsj86715/ingress/glyph/pages/GlyphPairsAdapter.kt
package com.hsj86715.ingress.glyph.pages

import com.hsj86715.ingress.glyph.tools.Utils
import com.hsj86715.ingress.glyph.view.BaseRecyclerAdapter

/**
 * Created by hushujun on 2017/4/18.
 */
class GlyphPairsAdapter(glyphPairs: Array<Array<String>>) : BaseRecyclerAdapter() {
    var mGlyphPairs: Array<Array<String>> = glyphPairs
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.glyphPreView.setSequences(mGlyphPairs[position])
        holder.itemView.setBackgroundColor(Utils.stringToColor(mGlyphPairs[position][0]))
        holder.glyphPreView.setHackSequenceListener(mClickedListener)
    }

    override fun getItemCount(): Int {
        return mGlyphPairs.size
    }
}